workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowFolderWatcher.java

```diff
@@ -139,8 +139,13 @@ private void addWorkflow(Path workflowFile) throws IOException, ProcessingExcept
       return;
     }
     Workflow workflow = SwadlParser.fromYaml(workflowFile.toFile());
-    workflowEngine.deploy(workflow);
-    deployedWorkflows.put(workflowFile, workflow.getId());
+    Object instance = workflowEngine.parseAndValidate(workflow);
+    if (workflow.isToPublish()) {
+      workflowEngine.deploy(workflow, instance);
+    } else if (deployedWorkflows.get(workflowFile) != null && deployedWorkflows.get(workflowFile).getRight()) {
+      workflowEngine.undeploy(deployedWorkflows.get(workflowFile).getLeft());
+    }
+    deployedWorkflows.put(workflowFile, Pair.of(workflow.getId(), workflow.isToPublish()));
```

Comment:
And here the worklowFile entry will get overridden. Maybe in the draft mode, we do nothing so we only keep the new file version in disk.